src/ess.jl

```diff
@@ -348,9 +358,18 @@ function ess_rhat(
             k += 2
         end
 
+        # for antithetic chains
+        # - reduce variance by averaging truncation to odd lag and truncation to next even lag
+        # - prevent negative ESS for short chains by ensuring τ is nonnegative
+        # See discussions in:
+        # - § 3.2 of Vehtari et al. https://arxiv.org/pdf/1903.08008v5.pdf
+        # - https://github.com/TuringLang/MCMCDiagnosticTools.jl/issues/40
+        # - https://github.com/stan-dev/rstan/pull/618
+        # - https://github.com/stan-dev/stan/pull/2774
+        τ = max(0, 2 * sum_pₜ + max(0, ρ_even) - 1)
```

Comment:
Is this line "correct", i.e., desired, even if we never run the `while` loop (i.e., if `maxlag <= 2`? In these cases it would simplify to `τ = max(0, 2 * sum_pₜ)` since `ρ_even = 1`. And in contrast to the other values we would use the last `ρ_odd`.

Reply:
No it's not. I've modified to compute the final `ρ_even` only if `maxlag >= 2`. Otherwise it is set to 0 before this line.